pulsar-broker/src/main/java/org/apache/pulsar/broker/service/schema/SchemaRegistryServiceImpl.java

```diff
@@ -181,56 +183,85 @@ public CompletableFuture<List<CompletableFuture<SchemaAndMetadata>>> getAllSchem
+                                    }
+                                    CompletableFuture<Void> checkCompatibilityFuture = new CompletableFuture<>();
+                                    if (schemaAndMetadataList.size() != 0) {
+                                        if (isTransitiveStrategy(strategy)) {
+                                            checkCompatibilityFuture = checkCompatibilityWithAll(schemaId, schema,
+                                                    strategy, schemaAndMetadataList);
+                                        } else {
+                                            checkCompatibilityFuture = checkCompatibilityWithLatest(schemaId,
+                                                    schema, strategy);
+                                        }
+                                    } else {
+                                        checkCompatibilityFuture.complete(null);
+                                    }
+                                    return checkCompatibilityFuture.thenCompose(v -> {
+                                        byte[] context = hashFunction.hashBytes(schema.getData()).asBytes();
+                                        SchemaRegistryFormat.SchemaInfo info = SchemaRegistryFormat.SchemaInfo
+                                                .newBuilder()
+                                                .setType(Functions.convertFromDomainType(schema.getType()))
+                                                .setSchema(ByteString.copyFrom(schema.getData()))
+                                                .setSchemaId(schemaId)
+                                                .setUser(schema.getUser())
+                                                .setDeleted(false)
+                                                .setTimestamp(clock.millis())
+                                                .addAllProps(toPairs(schema.getProps()))
+                                                .build();
+
+                                        long start = this.clock.millis();
+                                        schemaStorage
+                                                .put(schemaId, info.toByteArray(), context)
+                                                .thenAccept(sv -> {
+                                                    if (log.isDebugEnabled()) {
+                                                        log.debug("[{}] Schema is successfully added", schemaId);
+                                                    }
+                                                    this.stats.recordPutLatency(schemaId, this.clock.millis() - start);
+                                                    persistentFuture.complete(sv);
+                                                })
+                                                .exceptionally(t -> {
+                                                    if (t != null) {
```

Comment:
We do not need these two logic branch, right?

```java
.exceptionally(t ->{
  // the variable t will not be null
}
```

Reply:
You're right.